drafting.typ

```diff
@@ -192,53 +217,197 @@
+  }
+
+  return defaults
+}
+
 
 /// Required for `margin-note` to work, since it informs `drafting` of the page setup.
 #let set-page-properties(..kwargs) = {
   show: place // Avoid extra whitespace
   context {
     let kwargs = kwargs.named()
     layout(size => {
-      let margin = page.margin
-      if type(margin) != dictionary {
-        margin = (left: margin, right: margin)
-      }
-      if (auto in (margin.left, margin.right) and (page.width, page.height) == (auto, auto)) {
-        panic("If the page width *and* height are set to `auto`, neither left nor right margin" + " can be `auto`. Got (left, right) margin " + repr((
-          margin.left,
-          margin.right,
-        )) + ", and page (width, height) " + repr((page.width, page.height)))
-      }
-      // https://github.com/typst/typst/issues/3636#issuecomment-1992541661
-      let page-dims = (page.width, page.height).filter(x => x != auto)
-      let defaults = (
-        margin-left: margin.left,
-        margin-right: margin.right,
-      )
-      for (k, v) in defaults {
-        if v == auto {
-          defaults.at(k) = 2.5 / 21 * calc.min(..page-dims)
-        } else if type(v) == type(10pt - 100%) {
-          defaults.at(k) = v.length
-        }
+      let margins = _get-margins()
+
+      let binding = if page.binding == auto {
+        if text-direction(text.dir, text.lang) == ltr {
+          left
+        } else { right }
+      } else {
+        page.binding
       }
+
       set-margin-note-defaults(
-        ..defaults,
+        ..margins,
         page-width: size.width,
+        page-binding: binding,
         ..kwargs,
       )
     })
   }
 }
 
+#let _get-note-outline-props(note) = {
+  let func = note.func()
+  let defaults = margin-note-defaults.get()
+  // styled types from custom notes have unknown fill/stroke/body
+  // TODO: Maybe a better way to signify "unknown"?
+  let props = (
+    fill: note.at("fill", default: defaults.fill),
+    stroke: note.at("stroke", default: defaults.stroke),
+    body: note.at("body", default: "<body text unkonwn>"),
+  )
+
+  return props
+}
+
+/// Show an outline of all notes -> content
+#let note-outline(
+  /// Title of the outline -> string
+  title: "List of Notes",
+  /// Level of the heading -> number
+  level: 1,
+  /// Spacing between outline elements  -> relative
+  row-gutter: 10pt,
+) = context {
+  heading(level: level, title)
+
+  let notes = query(selector(<margin-note>).or(<inline-note>)).map(note => {
+    show: box // do not break entries across pages
+    let note-props = _get-note-outline-props(note)
+    let paint = stroke(note-props.stroke).paint
+    link(
+      note.location().position(),
+      grid(
+        columns: (1em, 1fr, 10pt),
+        column-gutter: 5pt,
+        align: (top, bottom, bottom),
+        box(
+          fill: note-props.fill,
+          stroke: if note-props.stroke == none {
+            none
+          } else if paint != note-props.fill {
```

Comment:
Maybe it is of lesser concern (as it seems weird to do), but this panics in the `stroke` constructor if `note-props.stroke` is `none`, which is possible (in the current version).

```typ
#set-margin-note-defaults(stroke: none)
```